src/types.ts

```diff
@@ -4,20 +4,40 @@
  */
 
 /* eslint-disable @typescript-eslint/no-empty-interface */
-
-import type * as express from 'express';
 import type * as http from 'http';
 import type * as httpProxy from 'http-proxy';
 import type * as net from 'net';
 import type * as url from 'url';
 
-export interface Request extends express.Request {}
-export interface Response extends express.Response {}
+export type Request = http.IncomingMessage;
+export type Response = http.ServerResponse;
 
-export interface RequestHandler extends express.RequestHandler {
-  upgrade?: (req: Request, socket: net.Socket, head: any) => void;
+/**
+ * http-proxy-middleware supports framework specific values. The following
+ * values are primarily decorated onto IncomingMessage by express, but are
+ * not required for use.
+ */
+declare module 'http' {
+  interface IncomingMessage {
+    originalUrl?: string;
+    hostname?: string;
+    host?: string;
+    body?: Record<string, any>;
+  }
 }
 
+export type Next = (...args: unknown[]) => unknown;
```

Comment:
```suggestion
export type Next = (err?: any) => void;
```

To match https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L34

Reply:
that's an express specific API, which this PR is intentionally trying to decouple from. using that type would assert that param0 is always an err in other server's next interface, which may not be the case. for max x-server compat, i'd opt to not offer such a coulped interface, but perhaps allow a user to extend somehow.

i have to refactor this for the v3 branch, i'll think more on it shortly